plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiUtil.java

```diff
@@ -149,11 +172,33 @@ public static List<HivePartitionKey> buildPartitionKeys(List<Column> keys, List<
         return partitionKeys.build();
     }
 
-    public static HoodieTableMetaClient buildTableMetaClient(HdfsEnvironment hdfsEnvironment, ConnectorSession session, String basePath)
+    // TODO: replace with buildTrinoHudiTableMetaClient
+    public static HoodieTableMetaClient buildTableMetaClient(
+            HdfsEnvironment hdfsEnvironment,
+            ConnectorSession session,
+            TrinoFileSystemFactory trinoFileSystemFactory,
+            String basePath)
+    {
+        HoodieTableMetaClient client = HoodieTableMetaClient.builder()
+                .setConf(hdfsEnvironment.getConfiguration(new HdfsContext(session), new Path(basePath)))
+                .setTrinoFileSystem(trinoFileSystemFactory.create(session))
+                .setBasePath(Location.of(basePath))
+                .build();
+        return client;
+    }
+
+    public static io.trino.plugin.hudi.table.HoodieTableMetaClient buildTrinoHudiTableMetaClient(
```

Comment:
Do we still need both versions of the method?

Reply:
no, one is removed now